rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+                if options is not None:
+
+                    options = dict(sorted(options.items()))
+                    param_str += '&'.join(  # params need to be sorted to be correctly validated
+                        [
+                            str(k) + '=' + str(v)
+                            for k, v in sorted(options.items())
+                            if v is not None
+                        ],
+                    )
+
+                signature = self._generate_signature(param_str=param_str)
+                self.session.headers = {
+                    'Content-Type': 'application/json',
+                    'X-BAPI-SIGN-TYPE': '2',
+                    'X-BAPI-RECV-WINDOW': self.recv_window,
+                    'X-BAPI-API-KEY': self.api_key,
+                    'X-BAPI-TIMESTAMP': str(timestamp),
+                    'X-BAPI-SIGN': signature,
+                }
+            else:
+                self.session.headers = {'Content-Type': 'application/json', 'X-BAPI-SIGN-TYPE': '2'}  # noqa: E501
+
+            try:
+                response = self.session.request(
+                    method='get',
+                    url=url,
+                    params=options,
+                    timeout=timeout,
+                )
+            except requests.exceptions.RequestException as e:
+                raise RemoteError(f'Bybit API request failed due to {e}') from e
+
+            if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                if tries >= 1:
+                    backoff_seconds = 10 / tries
+                    log.debug(
+                        f'Got a 429 from Bybit. Backing off for {backoff_seconds}')
+                    gevent.sleep(backoff_seconds)
+                    tries -= 1
+                    continue
+
+                # else
+                raise RemoteError(
+                    f'Bybit api request for {response.url} failed with HTTP '
+                    f'status code {response.status_code} and response {response.text}',
+                )
+
+            if response.status_code != HTTPStatus.OK:
+                raise RemoteError(
+                    f'Bybit api request for {response.url} failed with HTTP status '
+                    f'code {response.status_code} and response {response.text}',
+                )
+
+            break  # else all good, we can break off the retry loop
+
+        log.debug(f'BYBIT RESPONSE: {response.text}')
+        try:
+            json_ret = json.loads(response.text)
+        except json.JSONDecodeError as e:
+            raise RemoteError('Bybit returned invalid JSON response') from e
+
+        if 'result' not in json_ret:
+            raise RemoteError(f'Remote response is missing expected field result: {json_ret}')
+
+        if json_ret['retCode'] != 0 and json_ret['retMsg'] != 'OK':
+            raise RemoteError(f'Bybit returned error in request {json_ret}')
+
+        return json_ret['result']
```

Comment:
Here it can also raise KeyError the way you have it.

```suggestion
        if (result := json_ret.get('result')) is None:
            raise RemoteError(f'Remote response is missing expected field result: {json_ret}')

        if json_ret.get('retCode', 0) != 0 and json_ret.get('retMsg', 'OK') != 'OK':
            raise RemoteError(f'Bybit returned error in request {json_ret}')

        return result
```